sumcheck/src/prover_helper/arbitrary_gate.rs

```diff
@@ -0,0 +1,187 @@
+use core::panic;
+
+use arith::{Field, FieldSerde};
+use polynomials::{MultiLinearPoly, UnivariatePolynomial};
+use transcript::Transcript;
+
+#[derive(Debug, Clone)]
+pub struct SumcheckInstanceProof<F: Field> {
+    pub uni_polys: Vec<UnivariatePolynomial<F>>,
+}
+
+impl<F: Field + FieldSerde> SumcheckInstanceProof<F> {
+    pub fn new(uni_polys: Vec<UnivariatePolynomial<F>>) -> SumcheckInstanceProof<F> {
+        SumcheckInstanceProof { uni_polys }
+    }
+
+    /// Create a sumcheck proof for polynomial(s) of arbitrary degree.
+    ///
+    /// Params
+    /// - `claim`: Claimed sumcheck evaluation (note: currently unused)
+    /// - `num_rounds`: Number of rounds of sumcheck, or number of variables to bind
+    /// - `polys`: Dense polynomials to combine and sumcheck
+    /// - `comb_func`: Function used to combine each polynomial evaluation
+    /// - `transcript`: Fiat-shamir transcript
+    ///
+    /// Returns (SumcheckInstanceProof, r_eval_point, final_evals)
+    /// - `r_eval_point`: Final random point of evaluation
+    /// - `final_evals`: Each of the polys evaluated at `r_eval_point`
+    pub fn prove_arbitrary<Func, T>(
```

Comment:
a minor question regarding the naming - if I am guessing correctly, the naming suggests that it applies to arbitrary gate, i.e., the `comb_func` here.  In a sense, this also suggests specialized sumchecks will be (or have been) implemented for certain cases, say some low degree common cases.   Do these guesses sound reasonable to you?